nixos/modules/services/web-apps/matomo.nix

```diff
@@ -47,13 +51,27 @@ in {
         example = "lighttpd";
         # TODO: piwik.php might get renamed to matomo.php in future releases
         description = ''
-          Name of the web server user that forwards requests to the ${phpSocket} fastcgi socket for matomo if the nginx
+          Name of the web server user that forwards requests to the ${phpSocket} fastcgi socket for Matomo if the nginx
           option is not used. Either this option or the nginx option is mandatory.
           If you want to use another webserver than nginx, you need to set this to that server's user
           and pass fastcgi requests to `index.php` and `piwik.php` to this socket.
         '';
       };
 
+      periodicArchiveProcessing = mkOption {
+        type = types.bool;
+        default = true;
```

Comment:
Does it make sense for this to be true by default? Everybody using matomo will get it by default.

Reply:
I'd argue so, the only reason I can see to disable this is when a user wants to provide their own timer or use another method of execution instead of systemd timers.

The archive processing takes visits from the database and aggregates them to create plots and diagrams, it is non-destructive. If periodic archive processing is disabled, the same would happen when the user visits their matomo dashboard, just on demand instead of beforehand. On demand processing can even result in timeout, depending on the amount of data that has to be processed.

Even on low-end machines, e.g. Raspberry Pis, this makes sense, in order to make the dashboard more responsive and to split the work over time. With a Pi in mind though, I just added `CPUSchedulingPolicy = "idle"` and `IOSchedulingClass = "idle"` to reduce the responsiveness impact during the archive processing.